### PR TITLE
DTNOR-390: Complete release script

### PR DESCRIPTION
**Pull request by jcnoriega**
Ticket: https://engineering.paypalcorp.com/jira/browse/DTNOR-390

### Summary of changes
This PR adds the complete release script for Android. The script takes a version to release, and then executes the following steps:
- Build
- Unit Test each module (in parallel)
- Update versionName and versionCode in build.gradle
- Publish every module (in parallel)

- jcnoriega

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -104,6 +135,15 @@ jobs:
         with:
           signing_key_file: ${{ secrets.SIGNING_KEY_FILE }}
           signing_file_path: ${{ env.SIGNING_KEY_FILE_PATH }}
+      - name: Set github user
+        # We set these credentials because we have no way of getting the users (that started the workflow) email
+        # Reference: https://github.com/actions/checkout/issues/13
+        run: |
+          git config user.name github-actions[bot]
+          git config user.email 41898282+github-actions[bot]@users.noreply.github.com
+      - name: Pull updated files
```
Comment (sarahkoop) [inline .github/workflows/release.yml:136]: Do we need this pull step for any of these tasks? Unclear what we'd be pulling here, I thought this action runs off of whatever branch it's run from.

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -86,9 +91,35 @@ jobs:
...
+    name: Change Version
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v2
+      - name: Set github user
+          # We set these credentials because we have no way of getting the users (that started the workflow) email
+        # Reference: https://github.com/actions/checkout/issues/13
+        run: |
+          git config user.name github-actions[bot]
```
Comment (sarahkoop) [inline .github/workflows/release.yml]: Should we maybe put this as blocked until we can get an actual account for this? Also, I think we still need to update the Sonatype user in the secrets here to a group account. I don't think we'd want to publish using this script until both of those are complete.

Comment (jcnoriega) [reply .github/workflows/release.yml]: @sarahkoop the sonatype account is updated! its the user `braintree` in 1Password under Sonatype for Android publishing. Or do you mean creating a new one that its not `braintree`? perhaps a paypal one?

Comment (sarahkoop) [reply .github/workflows/release.yml]: Ah I missed that ok great!

File: .github/workflows/release.yml (hunk 1 above)
Comment (jcnoriega) [reply .github/workflows/release.yml:136]: @sarahkoop the workflow modifies the `build.gradle` to update the version name and code, then commits it and creates a tag. But the publishing is done on a diferent `job`. Each `job` runs on the state in which the branch is when the workflow starts (if a change a file in a `job`, that change is not translated to other `jobs`). That the reason why it pulls, to have the new `versionCode` and `versionName`

Comment (jcnoriega) [reply .github/workflows/release.yml:136]: `Each job runs in a fresh instance of the virtual environment specified by runs-on.` this phrase perhaps explains it better than me lol

**Review by andrespelaezp — APPROVED** [review]

File: .github/workflows/release.yml (hunk 1 above)
Comment (sarahkoop) [reply .github/workflows/release.yml:136]: That makes sense - thanks for explaining!

**Review by sarahkoop — APPROVED** [review]

**Pull request merged**